core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -301,9 +301,6 @@ private[spark] class DiskBlockManager(
    * Create a directory that is writable by the group.
    * Grant the permission 770 "rwxrwx---" to the directory so the shuffle server can
    * create subdirs/files within the merge folder.
-   * TODO: Find out why can't we create a dir using java api with permission 770
```

Comment:
For `TODO`, maybe we can see [https://bugs.openjdk.org/browse/JDK-8220013](https://bugs.openjdk.org/browse/JDK-8220013)

Reply:
Mind summarizing why we can use this approach while the JDK issue isn't resolved?

Reply:
In unix like systems every process has a property called umask which is masked onto the permissions of any file created and inherited by child processes. the default is 0002 or "turn off write for others". So it's most likely that Java is setting the permission you seek and then it's being masked out. At present, this issue has not been resolved, so my current approach is to create a directory first, and then explicitly grant permissions.